### PR TITLE
Fix corruption of FITS tables with unsigned int columns on save and load

### PR DESCRIPTION
**Pull request by eslavich**
Addresses #3655 

The problem on save turned out to be less of an astropy bug and more of a missing feature -- the public API doesn't currently allow us to put a FITS_rec into "pseudo unsigned integer" mode.  We were already messing with astropy implementation details by setting `_colrefs` directly, so I don't think it's a big deal to also set `_uint`.

The problem on load is, I think, an astropy bug.  I filed an issue and added a workaround here, which is to overwrite the incorrect FITS_rec dtype.

### CODE REVIEW

File: jwst/combine_1d/combine1d.py [hunk 1]
```diff
@@ -58,11 +58,6 @@ def __init__(self, ms, spec, exptime_key):
             self.surf_bright = np.zeros_like(self.flux)
             self.sb_error = np.zeros_like(self.flux)
             log.warning("There is no SURF_BRIGHT column in the input.")
-        # xxx temporary self.dq = spec.spec_table.field("dq").copy()
```
Comment (philhodge) [inline jwst/combine_1d/combine1d.py]: Only the "# xxx temporary " part of this line should be deleted.  The actual assignment to self.dq needs to be done.

File: jwst/datamodels/properties.py [hunk 2]
```diff
@@ -56,8 +56,12 @@ def _as_fitsrec(val):
         return val
     else:
         coldefs = fits.ColDefs(val)
+        uint = any(c._pseudo_unsigned_ints for c in coldefs)
         fits_rec = fits.FITS_rec(val)
         fits_rec._coldefs = coldefs
+        # FITS_rec needs to know if it should be operating in pseudo-unsigned-ints mode,
+        # otherwise it won't properly convert integer columns with TZEROn before saving.
+        fits_rec._uint = uint
```
Comment (philhodge) [inline jwst/datamodels/properties.py]: Is this used in astropy.io.fits?

File: jwst/combine_1d/combine1d.py (hunk 1 above)
Comment (eslavich) [reply jwst/combine_1d/combine1d.py]: Oh!  Oops, thanks.

File: jwst/datamodels/properties.py (hunk 2 above)
Comment (eslavich) [reply jwst/datamodels/properties.py]: Yes, [here](https://github.com/astropy/astropy/blob/de3933688433e22765b1bdffa175fcd954d691c3/astropy/io/fits/fitsrec.py#L940).  The else... is where it was changing the dtype to float64.

File: jwst/combine_1d/combine1d.py (hunk 1 above)
Comment (eslavich) [reply jwst/combine_1d/combine1d.py]: (this has been fixed)

**Comment by hbushouse** [comment]
Shall we go ahead and merge this?

**Pull request merged**